### PR TITLE
Add support to set the Slack URL in a file

### PR DESCRIPTION
**Pull request by julienduchesne**
- Added support for the file in both the global and the lower level
- Tried to follow configuration patterns I saw in prometheus
- The slack file is read on every request as mentioned in the prometheus issue to enable seamless switches

https://github.com/prometheus/alertmanager/issues/2498

### CODE REVIEW

File: config/testdata/conf.slack-no-api-url.yml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+route:
+  receiver: 'slack-notifications'
+  group_by: [alertname, datacenter, app]
+
+receivers:
+- name: 'slack-notifications'
+  slack_configs:
+  - channel: '#alerts'
+    text: 'test'
```
Comment (beorn7) [inline config/testdata/conf.slack-no-api-url.yml:9]: No newline at end of file.

File: config/testdata/conf.slack-default-api-url-file.yml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    text: 'test'
+  # Override global with other file
+  - channel: '#alerts2'
+    text: 'test'
+    api_url_file: '/override_file'
+  # Override global with inline URL
+  - channel: '#alerts3'
+    text: 'test'
+    api_url: 'http://mysecret.example.com/'
+
```
Comment (beorn7) [inline config/testdata/conf.slack-default-api-url-file.yml:22]: No newline at end of file. (But some whitespace on line 22, which just needs to be deleted?)

File: config/testdata/conf.slack-both-file-and-url.yml [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+route:
+  receiver: 'slack-notifications'
+  group_by: [alertname, datacenter, app]
+
+receivers:
+- name: 'slack-notifications'
+  slack_configs:
+  - channel: '#alerts1'
+    text: 'test'
+
```
Comment (beorn7) [inline config/testdata/conf.slack-both-file-and-url.yml:14]: No newline at end of file. (But some whitespace on line 22, which just needs to be deleted?)

**Review by beorn7 — APPROVED** [review]
I think this works. @roidelapluie @simonpasquier do you want to have a look, too?

**Comment by beorn7** [comment]
Of course, eventually we have to do the same for the other fields listed in #2498 . Not sure if we can reuse some code or just have to do it all individually.

**Review by beorn7 — CHANGES_REQUESTED** [review]
Just realized that docs/configuration.md needs to be updated, too. And perhaps an example in docs/notification_examples.md .

**Review by beorn7 — APPROVED** [review]
Thank you.

**Comment by beorn7** [comment]
@julienduchesne unfortunately, now the DCO is missing on a commit. See https://github.com/prometheus/alertmanager/pull/2534/checks?check_run_id=2332859284

**Comment by julienduchesne** [comment]
> @julienduchesne unfortunately, now the DCO is missing on a commit. See https://github.com/prometheus/alertmanager/pull/2534/checks?check_run_id=2332859284

Sorry about that. Fixed 👍

**Comment by beorn7** [comment]
Cool. Merging now, assuming that silence from @simonpasquier and @roidelapluie means consent. :->

**Pull request merged**